### PR TITLE
Chat adapter error update

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Fixes issue where adapter errors aren't being caught
# Why

Issue where `sendTypingIndicator` errors were not being caught by the adapter.

https://github.com/Azure/communication-ui-library/issues/2425
https://skype.visualstudio.com/SPOOL/_workitems/edit/3026772
# How Tested

Added new unit test to validate the error capture.

### CODE REVIEW

**Review by prprabhu-ms — APPROVED** [review]

File: packages/react-composites/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
     "build:e2e:call": "rushx _by-flavor \"webpack-cli build --config ./tests/app/call/webpack.config.js --mode=production --env production\"",
     "build:e2e:chat": "rushx _by-flavor \"webpack-cli build --config ./tests/app/chat/webpack.config.js --mode=production --env production\"",
     "build:e2e:callwithchat": "rushx _by-flavor \"webpack-cli build --config ./tests/app/callwithchat/webpack.config.js --mode=production --env production\"",
-    "test": "rushx _by-flavor \"rushx _test:by-flavor\"",
+    "test": "rush build -t . && rushx _by-flavor \"rushx _test:by-flavor\"",
```
Comment (PorterNan) [inline packages/react-composites/package.json:23]: I got the purpose of adding rush build -t here.
Somehow call rush build sounds like a parent level command

Imagine if you are doing a `rush test`, this `rush build -t` will run for every child repo which might cause some duplication, I guess we just need to give developers expectation that we don't do a full build for the test here

File: packages/chat-stateful-client/src/StatefulChatThreadClient.ts [hunk 2]
```diff
@@ -109,6 +109,14 @@ class ProxyChatThreadClient implements ProxyHandler<ChatThreadClient> {
       case 'listReadReceipts': {
         return createDecoratedListReadReceipts(chatThreadClient, this._context);
       }
+      case 'sendTypingNotification': {
```
Comment (PorterNan) [inline packages/chat-stateful-client/src/StatefulChatThreadClient.ts:112]: there is a unit test for statefulChatThreadClient:
StatefulCallClient.test.ts in the same folder

And there is a test set named 'errors should be reported correctly from Call when'

Would be nice to add this into the test set

**Review by PorterNan — APPROVED** [review]

File: packages/react-composites/package.json (hunk 1 above)
Comment (prprabhu-ms) [reply packages/react-composites/package.json:23]: Hmm.. Actually, I read this wrong earlier. `rushx build` would have been an acceptable trade-off. Running `rush build -t` when developer ran `rushx test` is wrong -- the expectation for a `rushx` command is that it is packlet-local only.

Comment (prprabhu-ms) [reply packages/react-composites/package.json:23]: Let's remove this from this PR and figure out if we need better documentation / understand why the test failed without the `rush build -t`.

**Pull request merged**